libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -148,6 +148,22 @@ def _convert_dict_to_message(_dict: Mapping[str, Any]) -> BaseMessage:
         return ChatMessage(content=_dict.get("content", ""), role=role, id=id_)
 
 
+def _format_message_content(content: Any) -> Any:
+    """Format message content."""
+    if content and isinstance(content, list):
+        # Remove unexpected block types
+        formatted_content = []
+        for block in content:
+            if isinstance(block, dict) and "type" in block and block["type"] != "text":
```

Comment:
openai client behaves OK when we pass in lists of dicts with `"type": "text"`. I was unable to find where the allowed dict structures are documented. https://github.com/openai/openai-python/blob/main/src/openai/types/chat/chat_completion_assistant_message_param.py#L30